tests/unit/test_command_install.py

```diff
@@ -94,3 +100,42 @@ def test_most_cases(
             lambda **kw: site_packages_writable
         )
         assert decide_user_install(use_user_site=None) is result
+
+
+def error_creation_helper(with_errno=False):
```

Comment:
This seems unnecessary.
You could directly provide `EnvironmentError("Some message)` or `EnvironmentError(errno.EACCES, "No file permission")` to the `parametrize` call below.